File: backend/src/main/kotlin/com/example/backend/models/CodePostal.kt
package com.example.backend.models

import jakarta.persistence.*

@Entity
@Table(name = "code_postal")
class CodePostal {
  @Id
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "code", length = 16)
  var code: String? = null

  @Column(name = "nom", nullable = false, length = 32)
  var nom: String? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "Ville_id")
  var ville: com.example.backend.models.Ville? = null
}

File: backend_py/pharmaPython/app/models/code_postal.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from app.core.db import Base

class CodePostal(Base):
  __tablename__ = "code_postal"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True, index=True)
  code = Column(String(16))
  nom = Column(String(32), nullable=False)
  ville_id = Column(Integer, ForeignKey("ville.id"))
  ville = relationship("Ville", lazy="joined")
